### PR TITLE
feat: updated to turbo v2

### PR DESCRIPTION
**Pull request by imCorfitz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,8 +67,9 @@
     "prettier-plugin-curly": "^0.2.1",
     "prettier-plugin-packagejson": "^2.5.0",
     "rimraf": "^5.0.7",
-    "turbo": "^1.13.3",
+    "turbo": "^2.0.3",
     "typescript": "^5.4.5",
     "yargs": "^17.7.2"
-  }
+  },
+  "packageManager": "bun@1.1.12"
```
Comment (ixahmedxi) [inline package.json:74]: does this actually work?

Comment (imCorfitz) [reply package.json:74]: This entire commit is the result of running `npx @turbo/codemod migrate` ... And all turbo commands continue to work. Removing the `packageManager` makes turbo bark at me.
Screenshot 2024-06-08 at 12 33 15

Comment (ixahmedxi) [reply package.json:74]: ohhh makes sense then.

**Review by ixahmedxi — DISMISSED** [review]

**Comment by imCorfitz** [comment]
I also added a fix for the `bun turbo lint` command, which failed for `@orbitkit/env` pacakge.
Screenshot 2024-06-09 at 14 24 56

File: packages/env/src/web.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const env = createEnv({
   },
   client: {},
   experimental__runtimeEnv: {
-    NODE_ENV: process.env['NODE_ENV'],
+    NODE_ENV: process.env.NODE_ENV,
   },
   emptyStringAsUndefined: true,
   skipValidation: !!process.env['SKIP_ENV_VALIDATION'],
```
Comment (imCorfitz) [inline packages/env/src/web.ts:33]: Curious why eslint doesn't have a problem with ['SKIP_ENV_VALIDATION']

Comment (ixahmedxi) [reply packages/env/src/web.ts:33]: because `NODE_ENV` is known to be on `process.env` but `SKIP_ENV_VALIDATION` isn't

**Review by ixahmedxi — APPROVED** [review]

**Pull request merged**